lib/replica_pools/query_cache.rb

```diff
@@ -19,12 +19,10 @@ module QueryCache
     # connection for cache logic, but ultimately pass its query
     # through to whatever connection is current.
     def select_all(*args)
-      # there may be more args for Rails 5.0+, but we only care about arel, name, and binds for caching.
       relation, name, raw_binds = args
+      arel = arel_from_relation(relation)
 
-      # Rails 6.2 breaks this method as locked? is no longer available
-      # https://github.com/kickstarter/replica_pools/issues/26
-      if !query_cache_enabled || locked?(relation)
+      if !query_cache_enabled && !(arel.respond_to?(:locked) && arel.locked)
```

Comment:
I think this isn't quite right, based on the test failure. I based it off of [the Rails 7 version of this method](https://github.com/rails/rails/blob/v7.0.3.1/activerecord/lib/active_record/connection_adapters/abstract/query_cache.rb#L96)

Reply:
I decided to pull back in the `locked?` method from Rails 6.1.